File: src/main/java/fr/lacombe/Quantity.java
package fr.lacombe;

public class Quantity {
    private double value;

    private Quantity(double value) {
        this.value = value;
    }

    public static Quantity valueOf(double value) {
        return new Quantity(value);
    }

    public double multiply(double price) {
        return price * this.value;
    }
}

File: CaisseEnregistreuse/CaisseEnregistreuse/Quantity.cs
namespace CashRegister
{
    public class Quantity
    {
        private readonly int Value;

        private Quantity(int value)
        {
            Value = value;
        }

        public static Quantity ValueOf(int value)
        {
            return new Quantity(value);
        }

        public static double operator *(double value, Quantity quantity)
        {
            return value * quantity.Value;
        }
    }
}